packages/fx-core/src/plugins/resource/frontend/utils/progress-helper.ts

```diff
@@ -4,131 +4,30 @@
 import { IProgressHandler, PluginContext } from "@microsoft/teamsfx-api";
 import { Messages } from "../resources/messages";
 
-export const ScaffoldSteps = {
-  Scaffold: Messages.ProgressScaffold,
-};
-
-export const PostProvisionSteps = {
-  EnableStaticWebsite: Messages.ProgressEnableStorageStaticWebsite,
-};
-
-export const PreDeploySteps = {
-  CheckStorage: Messages.ProgressCheckStorage,
-};
-
-export const DeploySteps = {
-  NPMInstall: Messages.ProgressNPMInstall,
-  Build: Messages.ProgressBuild,
-  getSrcAndDest: Messages.ProgressGetSrcAndDest,
-  Clear: Messages.ProgressClear,
-  Upload: Messages.ProgressUpload,
-};
-
-export const MigrateSteps = {
-  Migrate: Messages.ProgressMigrate,
-};
+export interface IProgress {
+  title: string;
+  steps: { [key: string]: string };
+}
 
 export class ProgressHelper {
-  static scaffoldProgress: IProgressHandler | undefined;
-  static postProvisionProgress: IProgressHandler | undefined;
-  static preDeployProgress: IProgressHandler | undefined;
-  static deployProgress: IProgressHandler | undefined;
-  static migrateProgress: IProgressHandler | undefined;
-
-  static async startScaffoldProgressHandler(
-    ctx: PluginContext
-  ): Promise<IProgressHandler | undefined> {
-    await this.scaffoldProgress?.end(true);
-
-    this.scaffoldProgress = ctx.ui?.createProgressBar(
-      Messages.ScaffoldProgressTitle,
-      Object.entries(ScaffoldSteps).length
-    );
-    await this.scaffoldProgress?.start(Messages.ProgressStart);
-    return this.scaffoldProgress;
-  }
+  static progressHandler: IProgressHandler | undefined;
 
-  static async startPostProvisionProgressHandler(
-    ctx: PluginContext
+  static async startProgress(
+    ctx: PluginContext,
+    progress: IProgress
   ): Promise<IProgressHandler | undefined> {
-    await this.postProvisionProgress?.end(true);
+    await this.progressHandler?.end(true);
 
-    this.postProvisionProgress = ctx.ui?.createProgressBar(
-      Messages.PostProvisionProgressTitle,
-      Object.entries(PostProvisionSteps).length
+    this.progressHandler = ctx.ui?.createProgressBar(
+      progress.title,
+      Object.entries(progress.steps).length
     );
-    await this.postProvisionProgress?.start(Messages.ProgressStart);
```

Comment:
Good to see this kind of refactor to make code clean and simple, thanks for your effort!